client/matching/isolation_loadbalancer.go

```diff
@@ -72,71 +76,50 @@ func (i *isolationLoadBalancer) PickWritePartition(taskListType int, req WriteRe
 
 func (i *isolationLoadBalancer) PickReadPartition(taskListType int, req ReadRequest, isolationGroup string) string {
 	taskList := *req.GetTaskList()
-	nRead := i.provider.GetNumberOfReadPartitions(req.GetDomainUUID(), taskList, taskListType)
-	taskListName := taskList.Name
 
-	if nRead <= 1 {
-		return taskListName
+	domainName, err := i.domainIDToName(req.GetDomainUUID())
+	if err != nil || !i.isolationEnabled(domainName) {
```

Comment:
Should we use fallback for zero isolation groups case instead?

Reply:
That was covered implicitly by `getPartitionsForGroup`, but I've made it more explicit.